src/app/components/calendar/weeklyCalendar.tsx

```diff
@@ -51,15 +57,15 @@ const WeeklyCalendar = ({ calendar, calendarSetup }: { calendar: any, calendarSe
     calendar.weekNumber = getWeekNumber(new Date());
   }
 
-  if (calendar.weekDayNumbers.length === 0) {
+  if (calendar.weekDayNumbers?.length === 0) {
     calendar.weekDayNumbers = defaultWeekDayNumbers();
   }
 
   let weekDays: number[] = fillWeekdays(calendar || defaultWeekDayNumbers(), []);
```

Comment:
Whoa, this is not intended, is it?

```suggestion
  let weekDays: number[] = fillWeekdays(calendar, []) || defaultWeekDayNumbers();
```

Adding a type to `fillWeekdays` would help discover this issue.

Relatedly, why is it even allowed for `weekDayNumbers` not to be present?